### DISCUSSION TITLE
New plugin/tool not showing up in the UI

### DISCUSSION THREAD

**Discussion by simonedstrom96** [danny-avila/LibreChat#3609] [Troubleshooting] [answered]
### What happened?

I followed the [tutorial for creating a new plugin/tool](https://www.librechat.ai/docs/development/tools_and_plugins) as closely as I could but when launching the app the new plugin/tool is not seen anywhere. [Here are the source code changes](https://github.com/danny-avila/LibreChat/compare/main...simonedstrom96:LibreChat:add_react_agent). I have verified that other tools like google search works.

### Steps to Reproduce

1. Create a simple custom agent/tool according to the provided documentation.
2. Try to find the tool among the plugins or in the Assistants tools
3. See that its not there and therefore cannot be selected.

### What browsers are you seeing the problem on?

Chrome

### Screenshots

![image](https://github.com/user-attachments/assets/9cfa2b59-e610-4bc7-b73d-32dee4ec1be7)
![image](https://github.com/user-attachments/assets/8fac3229-2e29-4b43-97bc-c8ce6fd2caaa)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
If you are using docker or containers, you may have to mount the `api/app/clients/tools/manifest.json` file to your API container for changes to be seen.

**Reply by danny-avila** [reply]
For example, I only edited the top of `api/app/clients/tools/manifest.json`:

```json
[
  {
    "name": "Super Cool Tool",
    "pluginKey": "super_cool_tool",
    "description": "This tool rocks",
    "icon": "https://tool.ai/favicon.ico",
    "authConfig": [
      {
        "authField": "TOOL_API_KEY",
        "label": "Tool API Key",
        "description": "Super cool, a tool"
      }
    ]
  },
  {
  ```
  
![image](https://github.com/user-attachments/assets/7aefde9e-6ab7-45bb-ab42-fd524907b0ad)

**Reply by danny-avila** [reply, marked as answer]
I would make use of a [Docker override file](https://www.librechat.ai/docs/configuration/docker_override), and add the following:


```yaml
services:
  api:
    volumes:
      - type: bind
        source: ./api/app/clients/tools/manifest.json
        target: /app/api/app/clients/tools/manifest.json
```

**Comment by simonedstrom96** [comment]
Thanks @danny-avila ! The docker override solved the problem. Perhaps this should be added to the docs for creating custom tools/plugins?

**Comment by simonedstrom96** [comment]
Unfortunately I'm now getting another error: "error: [PluginsClient] executorCall error: Cannot read properties of null (reading 'call')" with no further information in the logs that I can find. I do not think it is an inner logic problem since I have tried quite a few fixes but nothing seems to work.
